components/aero_sensors/FS/fatfs.c

```diff
@@ -0,0 +1,633 @@
+/**
+ * @file fatfs.h
+ * @brief  Source code for the FatFS interface
+ * @author Joshua Lafleur (josh.lafleur@outlook.com)
+ * @version 0.1
+ * @date 2022-07-16
+ *
+ * @note
+ * Portions copyright (C) 2014, ChaN, all rights reserved.
+ * Portions copyright (C) 2017, kiwih, all rights reserved.
+ *
+ * This software is a free software and there is NO WARRANTY.
+ * No restriction on use. You can use, modify and redistribute it for
+ * personal, non-profit or commercial products UNDER YOUR RESPONSIBILITY.
+ * Redistributions of source code must retain the above copyright notice.
+ *
+ ******************************************************************************
+ */
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+#include "fatfs.h"
+#include "ff_gen_drv.h"
+#include "integer.h"
+
+#include "HW_spi.h"
+
+
+/******************************************************************************
+ *                              D E F I N E S
+ ******************************************************************************/
+
+#define SD_SPI_INTERFACE ((SPI_TypeDef*)SPI2) /**< SD Interface on SPI2 */
+#define SD_INIT_TIMEOUT  1000
+#define SD_WAIT_TIMEOUT  500
+#define SD_RCVR_TIMEOUT  200
+
+#define FCLK_SLOW()                                                            \
+ {                                                                             \
+  MODIFY_REG(spi2->CR1, SPI_BAUDRATEPRESCALER_256, SPI_BAUDRATEPRESCALER_128); \
+ } /* Set SCLK = slow, 250 KBits/s*/
+#define FCLK_FAST()                                                          \
+ {                                                                           \
+  MODIFY_REG(spi2->CR1, SPI_BAUDRATEPRESCALER_256, SPI_BAUDRATEPRESCALER_8); \
+ } /* Set SCLK = fast, 4 MBits/s */
```

Comment:
Why are these spanning three lines each?

Reply:
clang format does this